p2p/simulations/adapters/exec.go

```diff
@@ -71,11 +69,11 @@ func (e *ExecAdapter) NewNode(config *NodeConfig) (Node, error) {
 	conf.P2P.ListenAddr = "127.0.0.1:0"
 
 	node := &ExecNode{
-		ID:      config.Id,
-		Service: config.Service,
-		Dir:     dir,
-		Config:  &conf,
-		key:     config.PrivateKey,
+		ID:          config.Id,
+		Service:     config.Service,
+		Dir:         dir,
+		StackConfig: &conf,
```

Comment:
I have reduced the separation by setting `ExecNode.Config` to `*execNodeConfig` which wraps both the stack and node config, better?

https://github.com/ethersphere/go-ethereum/blob/d7eb6b5a70d4f49887ce982068007821dad8b407/p2p/simulations/adapters/exec.go#L93